presto-product-tests/src/main/java/io/prestosql/tests/hive/HiveVersion.java

```diff
@@ -34,8 +39,22 @@ public int getMinorVersion()
         return minorVersion;
     }
 
-    public static HiveVersion createFromParts(int majorVersion, int minorVersion)
+    public int getPatchVersion()
+    {
+        return patchVersion;
+    }
+
+    private static final Pattern HIVE_VERSION_PATTERN = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+).*$");
```

Comment:
- can we expect `\\.` or `(\\.|$)` after 3rd number? (or how would we parse `3.1.2a3-xyz`?)
- move to the top
- technically, `^` and `$` are redundant (you use `matcher.matches()` anyway)

Reply:
`3.1.2a3-xyz` good point (maybe) that would fail of course. Though I did not ever see any letters in version string. returned by metadata (what I saw always looked like this: `3.1.3000.7.1.1.0-565`)
Let's address it when we spot it.